index.js

```diff
@@ -51,17 +51,23 @@ function init(config, logger, callback) {
 
     var conns = _.keys(cfg.connections);
     debug('Initializing connections ' +  conns);
-    async.forEach(conns, _initConnection, function (err) {
-        return callback(err);
+    var initConnPromises = [];
+    conns.forEach(function (conn) {
+        initConnPromises.push(_initConnection(conn));
+    });
+    Q.all(initConnPromises).then(function () {
```

Comment:
I knew you would ask about this : ). When I thought about it I determined that while we want to wait for all promises to settle in the updateDesigns case, this is not necessarily the case for the other async operations (verifying connections, databases) where it may make more sense to fail fast.